contracts/vaults/Claim.sol

```diff
@@ -0,0 +1,192 @@
+    * pay out bounty to hacker and committee. Also transfer to the 
+    * HATVaultsRegistry the part of the bounty that will be swapped to HAT 
+    * tokens.
+    * If the claim had been previously challenged, this is only callable by
+    * the arbitrator. Otherwise, callable by anyone after challengePeriod had
+    * passed.
+    * @param _claimId The claim ID
+    * @param _bountyPercentage The percentage of the vault's balance that will
+    * be sent as a bounty. This value will be ignored if the caller is not the
+    * arbitrator.
+    */
+    function approveClaim(bytes32 _claimId, uint256 _bountyPercentage) external nonReentrant isActiveClaim(_claimId) {
+        Claim memory claim = activeClaim;
+        delete activeClaim;
+
+        if (claim.challengedAt != 0) {
+            // solhint-disable-next-line not-rely-on-time
+            if (msg.sender != registry.arbitrator() || block.timestamp > claim.challengedAt + registry.challengeTimeOutPeriod())
+                revert ChallengedClaimCanOnlyBeApprovedByArbitratorUntilChallengeTimeoutPeriod();
+            claim.bountyPercentage = _bountyPercentage;
+        } else {
+            // solhint-disable-next-line not-rely-on-time
+            if (block.timestamp <= claim.createdAt + registry.challengePeriod()) revert UnchallengedClaimCanOnlyBeApprovedAfterChallengePeriod();
+        }
+
+        address tokenLock;
+
+        ClaimBounty memory claimBounty = calcClaimBounty(claim.bountyPercentage);
+
+        IERC20 asset = IERC20(asset());
+        if (claimBounty.hackerVested > 0) {
+            //hacker gets part of bounty to a vesting contract
+            tokenLock = tokenLockFactory.createTokenLock(
+                address(asset),
+                0x000000000000000000000000000000000000dEaD, //this address as owner, so it can do nothing.
+                claim.beneficiary,
+                claimBounty.hackerVested,
+                // solhint-disable-next-line not-rely-on-time
+                block.timestamp, //start
+                // solhint-disable-next-line not-rely-on-time
+                block.timestamp + vestingDuration, //end
+                vestingPeriods,
+                0, //no release start
+                0, //no cliff
+                ITokenLock.Revocability.Disabled,
+                false
+            );
+            asset.safeTransfer(tokenLock, claimBounty.hackerVested);
+        }
+
+        asset.safeTransfer(claim.beneficiary, claimBounty.hacker);
+        asset.safeTransfer(claim.committee, claimBounty.committee);
+
+        // send to the registry the amount of tokens which should be swapped 
+        // to HAT so it could call swapAndSend in a separate tx.
+        asset.safeApprove(address(registry), claimBounty.hackerHatVested + claimBounty.governanceHat);
+        registry.addTokensToSwap(
+            asset,
+            claim.beneficiary,
+            claimBounty.hackerHatVested,
+            claimBounty.governanceHat
+        );
+
```

Comment:
sets do an asset.safeApporve(registyr, 0) here just to be sure

Reply:
https://github.com/hats-finance/hats-contracts/pull/307